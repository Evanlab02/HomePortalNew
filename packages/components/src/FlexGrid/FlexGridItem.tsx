import { PropsWithChildren } from "react";

export interface FlexGridItemProps extends PropsWithChildren {
    xs?: number
    sm?: number
    md?: number
    lg?: number
    xl?: number
    xxl?: number
}

export default function FlexGridItem(props: Readonly<FlexGridItemProps>) {
	const { children, xs, sm, md, lg, xl, xxl } = props;

	const classes = [
		xs && `hp-flex-grid-col-xs-${xs.toString()}`,
		sm && `hp-flex-grid-col-sm-${sm.toString()}`,
		md && `hp-flex-grid-col-md-${md.toString()}`,
		lg && `hp-flex-grid-col-lg-${lg.toString()}`,
		xl && `hp-flex-grid-col-xl-${xl.toString()}`,
		xxl && `hp-flex-grid-col-xxl-${xxl.toString()}`
	].filter(Boolean).join(" ");

	return (
		<div className={classes} data-testid="hp-flex-grid-item">
			{children}
		</div>
	);
}