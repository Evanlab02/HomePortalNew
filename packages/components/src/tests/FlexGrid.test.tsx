import { render } from "@testing-library/react";
import { expect, test } from "vitest";
import FlexGridContainer from "../FlexGrid";
import FlexGridItem from "../FlexGrid/FlexGridItem";

test("Flex grid renders correctly", async () => {
	const { findByTestId } = render(
		<FlexGridContainer>
			<FlexGridItem xs={1} sm={2} md={3} lg={4} xl={5} xxl={6}>
				<p>Hello World!</p>
			</FlexGridItem>
		</FlexGridContainer>
	);

	const flexContainer = await findByTestId("hp-flex-grid-container");
	expect(flexContainer).toMatchSnapshot();
});