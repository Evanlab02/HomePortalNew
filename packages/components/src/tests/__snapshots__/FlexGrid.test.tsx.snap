// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Flex grid renders correctly 1`] = `
<div
  class="hp-flex-grid-container"
  data-testid="hp-flex-grid-container"
>
  <div
    class="hp-flex-grid-col-xs-1 hp-flex-grid-col-sm-2 hp-flex-grid-col-md-3 hp-flex-grid-col-lg-4 hp-flex-grid-col-xl-5 hp-flex-grid-col-xxl-6"
    data-testid="hp-flex-grid-item"
  >
    <p>
      Hello World!
    </p>
  </div>
</div>
`;
